### PR TITLE
closes #358 Ice codes

### LINKED ISSUE #358: Ice Code/Baltic Sea Ice Code
(Hendrik via FB)

https://www.code-knacker.de/eiscode.htm?fbclid=IwAR3A7TfLL2lN-qxVVzGT4FnHj28kYkeAXdGzUtUr29FDsccp3bOIZXdeoeI

### PR DESCRIPTION
**Pull request by t-m-z**
provides data from https://www.code-knacker.de/eiscode.htm in de, en, fr

 EU     https://eur-lex.europa.eu/legal-content/DE/TXT/PDF/?uri=CELEX:32007R0416
        S.  83 Ice condition code
        S.  89 Ice_accessibility_code
        S.  92 Ice_classification_code

 BALTIC https://www.bsh.de/EN/TOPICS/Marine_environment/Ice/Ice_observations/Ice_observations_node.html
        https://www.bsh.de/DE/THEMEN/Meeresumwelt/Eis/Eisbeobachtungen/eisbeobachtungen_node.html

 WMO    https://library.wmo.int/doc_num.php?explnum_id=4651
        S. 115 Concentration
        S. 116 Stage of development and thickness - Symbols
        S. 117 Form of ice
        S. 118 Stage of melting (ms)
        S. 119 Snow depth (s)

 SIGRID http://www.aari.ru/gdsidb/format/sigrid-1.pdf
        S. 16 Concentration

### CODE REVIEW

**Comment by S-Man42** [comment]
This branch cannot be merged due to conflicts in en.json...

I fear it is not longer possible to fill out the language files manually... It would always yield a problem when using Crowdin in parallel...

**Comment by S-Man42** [comment]
I really don't think, it's necessary to add DE/EN/FR choice on that. It could be done with setting the language file...

I believe you idea is: If an English speaker wants to solve a German Ice Code mystery, he should be able to access the German works. However, I guess, that's quite an unusual use case and makes it much more difficult to track all of our translations. If an English speaker finds a German Enigma cache, he must switch the language as well to find the input field for "Walze" instead of "Rotor".

I'd appreciate if you could remove this...

File: assets/i18n/de.json [hunk 1]
```diff
@@ -1568,6 +1568,30 @@
...
+  "icecodes_online_help_url" : "",
+  "icecodes_symbol" : "Eiszustandscodes",
+  "icecodes_selection_title" : "Eiszustandscodes",
+  "icecodes_selection_description" : "Eiszustandscodes auf Gewässern",
+  "icecodes_selection_online_help_url" : "https://blog.gcwizard.net/manual/de/eis-codes/06-was-zeigt-der-gc-wizard-an/",
+  "icecodes_system_eu" : "EG Verordnung Nr. 416/2007 vom 22.03.2007",
+  "icecodes_system_eu_condition" : "Eiszustandscode",
+  "icecodes_system_eu_accessibility" : "Eiszugangscode",
+  "icecodes_system_eu_classification" : "Eisbefahrbarkeitsccode",
+  "icecodes_system_baltic" : "Ostesse-Eiscode 1980",
```
Comment (S-Man42) [inline assets/i18n/de.json:1583]: Typo

File: assets/i18n/en.json [hunk 2]
```diff
@@ -1568,6 +1568,30 @@
   "humidex_index_30" : "Some discomfort.",
   "humidex_index_40" : "Great discomfort; avoid exertion.",
   "humidex_index_45" : "Dangerous; heat stroke quite possible.",
+  "icecodes_title" : "Ice codes",
```
Comment (S-Man42) [inline assets/i18n/en.json:1571]: Please make titles always with capital letters to be consistent throughout the app :) "Ice Codes"

File: assets/i18n/en.json [hunk 3]
```diff
@@ -1568,6 +1568,30 @@
...
+  "icecodes_description" : "Description of ice situation on waters",
+  "icecodes_example" : "1 → Light nilas or ice rind ( 5-10 cm thick)",
+  "icecodes_online_help_url" : "",
+  "icecodes_symbol" : "Ice codes",
+  "icecodes_selection_title" : "Ice codes",
+  "icecodes_selection_description" : "Description of ice situation on waters",
+  "icecodes_selection_online_help_url" : "https://blog.gcwizard.net/manual/en/ice-codes/06-what-shows-the-gc-wizard/",
+  "icecodes_system_eu" : "EU Regulation Nr. 416/2007 as of 22.03.2007",
+  "icecodes_system_eu_condition" : "Ice condition code",
```
Comment (S-Man42) [inline assets/i18n/en.json:1580]: Capital letters :)

File: assets/i18n/en.json [hunk 4]
```diff
@@ -1568,6 +1568,30 @@
...
   "humidex_index_45" : "Dangerous; heat stroke quite possible.",
+  "icecodes_title" : "Ice codes",
+  "icecodes_description" : "Description of ice situation on waters",
+  "icecodes_example" : "1 → Light nilas or ice rind ( 5-10 cm thick)",
+  "icecodes_online_help_url" : "",
+  "icecodes_symbol" : "Ice codes",
+  "icecodes_selection_title" : "Ice codes",
+  "icecodes_selection_description" : "Description of ice situation on waters",
+  "icecodes_selection_online_help_url" : "https://blog.gcwizard.net/manual/en/ice-codes/06-what-shows-the-gc-wizard/",
+  "icecodes_system_eu" : "EU Regulation Nr. 416/2007 as of 22.03.2007",
```
Comment (S-Man42) [inline assets/i18n/en.json:1579]: Maybe another date format? DD.MM.YYYY is German.

File: assets/i18n/fr.json [hunk 5]
```diff
@@ -1444,6 +1444,30 @@
...
   "humidex_index_45": "Dangereux; coup de chaleur tout à fait possible.",
+  "icecodes_title" : "Codes de la glace",
+  "icecodes_description" : "Description de la situation de la glace sur les eaux",
+  "icecodes_example" : "1 → Nilas léger ou croûte de glace ( 5-10 cm d'épaisseur)",
+  "icecodes_online_help_url" : "https://blog.gcwizard.net/manual/fr/code-de-glace/06-quest-ce-que-gc-wizard-affiche/",
+  "icecodes_symbol" : "Codes de condition de glace",
+  "icecodes_selection_title" : "Codes de condition de glace",
+  "icecodes_selection_description" : "Description de la situation de la glace sur les eaux",
+  "icecodes_selection_online_help_url" : "",
+  "icecodes_system_eu" : "Règlement UE n° 416/2007 du 22.03.2007",
```
Comment (S-Man42) [inline assets/i18n/fr.json:1455]: Frensh Date format?

File: lib/widgets/searchstrings/searchstrings_en.dart [hunk 6]
```diff
@@ -181,6 +181,7 @@ final SEARCHSTRING_EN_HASHES_TIGER_192 = '';
 final SEARCHSTRING_EN_HASHES_WHIRLPOOL_512 = '';
 final SEARCHSTRING_EN_HEXADECIMAL = 'hexadecimal decimal';
 final SEARCHSTRING_EN_HOMOPHONE = 'monoalphabetical letterfrequency';
+final SEARCHSTRING_EN_ICECODES = 'balticsea ice floe melting snow iceberg';
```
Comment (S-Man42) [inline lib/widgets/searchstrings/searchstrings_en.dart:184]: Please add "icecodes"

**Comment by t-m-z** [comment]
I am quite not sure  that this is ok. 
If I connect the output to the locale you will get the german translation only with de-settings. Hence an english or french has to change his settings to de to compare a german cache with Ice codes.
Currently the tool is similar to numeral words where we also allow different languages independent from the locale settings.

Sent from MailDroid

-----Original Message-----

**Comment by t-m-z** [comment]
I understand. So we have to agree to a certain procedure how to include the crowdin files and how to add strings to crowdin.
And I need probably some help on that.

Sent from MailDroid

-----Original Message-----

**Comment by t-m-z** [comment]
> This branch cannot be merged due to conflicts in en.json...
> 
> I fear it is not longer possible to fill out the language files manually... It would always yield a problem when using Crowdin in parallel...

This is something we are already discussing. But I am not convinced that merge-conflicts can be avoided

File: assets/i18n/de.json (hunk 1 above)
Comment (t-m-z) [reply assets/i18n/de.json:1583]: done

File: assets/i18n/en.json (hunk 2 above)
Comment (t-m-z) [reply assets/i18n/en.json:1571]: done

File: assets/i18n/en.json (hunk 3 above)
Comment (t-m-z) [reply assets/i18n/en.json:1580]: done - maybe we should have something like naming conventions.

File: assets/i18n/en.json (hunk 4 above)
Comment (t-m-z) [reply assets/i18n/en.json:1579]: Finaly found the document in the Web and correct the title

File: assets/i18n/fr.json (hunk 5 above)
Comment (t-m-z) [reply assets/i18n/fr.json:1455]: Finaly found the document in the Web and correct the title

File: lib/widgets/searchstrings/searchstrings_en.dart (hunk 6 above)
Comment (t-m-z) [reply lib/widgets/searchstrings/searchstrings_en.dart:184]: done

**Comment by t-m-z** [comment]
the merge conflict is resolved.
But I would like to stick to the option for different outputs independent from the locale-setting.
Move the different translations into the json-files would limit the usability.
However it is possible.

**Comment by t-m-z** [comment]
> I really don't think, it's necessary to add DE/EN/FR choice on that. It could be done with setting the language file...
> 
> I believe you idea is: If an English speaker wants to solve a German Ice Code mystery, he should be able to access the German works. However, I guess, that's quite an unusual use case and makes it much more difficult to track all of our translations. If an English speaker finds a German Enigma cache, he must switch the language as well to find the input field for "Walze" instead of "Rotor".
> 
> I'd appreciate if you could remove this...

Done

**Comment by S-Man42** [comment]
Manually merged in 

https://github.com/S-Man42/GCWizard/commit/f52c7dabda4cd1564a9c1a1d8cbfb44d84a6b4b5
https://github.com/S-Man42/GCWizard/commit/8652e4d358eaa47c4649fcc902cb2a3fbb5387fd

**Pull request closed without merging**